docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/CubeDockerConfigurator.java

```diff
@@ -87,6 +95,57 @@ private void configure(ArquillianDescriptor arquillianDescriptor) {
         return cubeConfiguration;
     }
 
+    private Map<String, String> resolveDownloadDockerMachine(Map<String, String> config) {
+        if (config.containsKey(CubeDockerConfiguration.DOCKER_MACHINE_NAME)) {
+            String machineVersion = null;
+            String machineArquillianPath = null;
+            File dockerMachineFile = null;
+            if (config.containsKey(CubeDockerConfiguration.DOCKER_MACHINE_VERSION)) {
+                machineVersion = config.get(CubeDockerConfiguration.DOCKER_MACHINE_VERSION);
+                machineArquillianPath = CubeDockerConfiguration.resolveMachinePath(machineVersion);
+                dockerMachineFile = new File(machineArquillianPath);
+            }
+            boolean dockerMachineFileExist = dockerMachineFile != null && dockerMachineFile.exists();
+            final String cliPathExec = config.get(CubeDockerConfiguration.DOCKER_MACHINE_PATH);
+            if (!dockerMachineInstance.get().isDockerMachineInstalled(cliPathExec) && !dockerMachineFileExist) {
+                String machineName = config.get(CubeDockerConfiguration.DOCKER_MACHINE_NAME);
+                String machineUrl = CubeDockerConfiguration.resolveUrl(machineVersion);
+
+                Spacelift.task(DownloadTool.class)
+                            .from(machineUrl)
+                            .to(dockerMachineFile)
+                            .execute()
+                            .await();
+                config.put(CubeDockerConfiguration.DOCKER_MACHINE_PATH, dockerMachineFile.getAbsolutePath());
+
+                Command allowExecCmd = new CommandBuilder("chmod", "+x", machineArquillianPath).build();
+                ProcessResult processAllowResult = Spacelift.task(CommandTool.class)
```

Comment:
maybe you could check that the chmod exited with exit code 0, there is API for that if you look more into it, via ProcessResult#exitValue and CommandTool#shouldExitWith